### PR TITLE
Added signtool.exe path in SecurityUtil.SigntoolNotFound error, fix #425

### LINKED ISSUE #425: Error when signtool.exe can't be found should include the path we looked in
[This line](https://github.com/Microsoft/msbuild/blob/98d38cbf34fe35142792556aff80cec08d7315cc/src/XMakeTasks/ManifestUtil/SecurityUtil.cs#L773) makes it hard to figure out *why* signtool appeared to be missing. We had an internal bug where the .exe was actually not present on disk, but the error was just:

```
1>Task "SignFile" (TaskId:43)
1>  Task Parameter:CertificateThumbprint=AAD0491335D003B4BAE60CCCEE28EC62600D224D (TaskId:43)
1>  Task Parameter:SigningTarget=bin\Debug\app.publish\\setup.exe (TaskId:43)
1>C:\Program Files (x86)\MSBuild\Microsoft\VisualStudio\v14.0\OfficeTools\Microsoft.VisualStudio.Tools.Office.targets(609,9): error MSB3482: An error occurred while signing: SignTool.exe not found.
1>Done executing task "SignFile" -- FAILED. (TaskId:43)
```

Would have been nice to know where it was looking!

### PR DESCRIPTION
**Pull request by nikson**
Added search path in `SecurityUtil.SigntoolNotFound` message, So it'll be easier to locate the exact path. For proper messaging move the `ApplicationException` to `GetPathToTool` method.

### CODE REVIEW

File: src/XMakeTasks/ManifestUtil/SecurityUtil.cs [hunk 1]
```diff
@@ -760,18 +758,20 @@ internal static string GetCommandLineParameters(string certThumbprint, Uri times
             return commandLine.ToString();
         }
 
-        internal static string GetPathToTool()
+        internal static string GetPathToTool(System.Resources.ResourceManager resources)
         {
-#pragma warning disable 618 // Disabling warning on using internal ToolLocationHelper API. At some point we should migrate this.
-            string toolPath = ToolLocationHelper.GetPathToWindowsSdkFile(ToolName, TargetDotNetFrameworkVersion.VersionLatest, VisualStudioVersion.VersionLatest);
+#pragma warning disable 618 // Disabling warning on using internal ToolLocationHelper API. At some point we should migrate this.            
+            string toolPath = ToolLocationHelper.GetPathToWindowsSdkFile(ToolName, TargetDotNetFrameworkVersion.VersionLatest, VisualStudioVersion.VersionLatest);
```
Comment (rainersigwald) [inline src/XMakeTasks/ManifestUtil/SecurityUtil.cs]: nit: trailing whitespace on this line and the previous one

File: src/XMakeTasks/ManifestUtil/Strings.resx [hunk 2]
```diff
@@ -145,7 +145,7 @@
     <value xml:space="preserve">Warning while signing {0}. {1}</value>
   </data>
   <data name="SecurityUtil.SigntoolNotFound">
-    <value xml:space="preserve">SignTool.exe not found.</value>
+    <value xml:space="preserve">SignTool.exe not found. Path: {0}</value>
```
Comment (rainersigwald) [inline src/XMakeTasks/ManifestUtil/Strings.resx]: Can you rephrase this to the slightly more natural `SignTool.exe was not found at path {0}`?

**Comment by rainersigwald** [comment]
Looks like a good change to me, if you address my nitpicks. Thanks!

File: src/XMakeTasks/ManifestUtil/Strings.resx (hunk 2 above)
Comment (nikson) [inline src/XMakeTasks/ManifestUtil/Strings.resx]: Rephrase should contain `?` mark?

Comment (rainersigwald) [inline src/XMakeTasks/ManifestUtil/Strings.resx]: No, but you're right that it should probably have *some* punctuation. Looks like we'd match the pattern in this file with

```
SignTool.exe was not found at path {0}.
```

**Comment by nikson** [comment]
Updated, @rainersigwald please have a look

**Comment by rainersigwald** [comment]
LGTM. Thanks!

**Comment by AndyGerlicher** [comment]
LGTM.
